### PR TITLE
PR of Experimental Branch

### PR DESCRIPTION
**Pull request by steevaavoo**
Fixed pester tests (thank you Adam!), removed redundant help uris, removed unnecessary parameter sets.

### CODE REVIEW

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 1]
```diff
@@ -46,24 +46,23 @@ function Get-sbRDSession {
 
     [OutputType('Custom.SB.RDSession')]
     param(
-        [cmdletbinding(DefaultParameterSetName = 'State')]
-        [Parameter(Mandatory = $false, ParameterSetName = 'State')]
+        [cmdletbinding()]
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: casing should be `CmdletBinding`

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 2]
```diff
@@ -46,24 +46,23 @@ function Get-sbRDSession {
...
         [string]$SessionState = "Any",
 
         # Switch to choose to include self - as we probably don't want to disconnect/logoff our own session, but
-
         # might want to test a message as an example - however, using the UserName parameter will allow current
         # user to be returned - which makes sense to me
-        [Parameter(Mandatory = $false, ParameterSetName = 'State')]
+        [Parameter(Mandatory = $false)]
         [Alias("Self")]
         [switch]$IncludeSelf = $false,
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: `$false` is the default for `[Switch]` anyway, so this is superfluous; like your face!

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 3]
```diff
@@ -90,43 +89,60 @@ function Get-sbRDSession {
...
-                $_.SessionState -like $stateLookup.$SessionState`
-                    -and ( $_.IdleTime / 60000 ) -ge $MinimumIdleMins`
+            }#foreach user
+        }
+        else {
+            if ($IncludeSelf) {
+                Write-Verbose "Querying RD Session Collection for [$SessionState] sessions - including [$env:USERNAME]"
+                $sessions = Get-RDUserSession | Where-Object {
+                    $_.SessionState -like $stateLookup.$SessionState`
+                        -and ( $_.IdleTime / 60000 ) -ge $MinimumIdleMins`
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: Only indent one level when lines run over

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 4]
```diff
@@ -90,43 +89,60 @@ function Get-sbRDSession {
...
+                }
+            }
+            else {
+                Write-Verbose "Querying RD Session Collection for [$SessionState] sessions"
+                $sessions = Get-RDUserSession | Where-Object {
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: Only indent one level, like this:
```     
$sessions = Get-RDUserSession | Where-Object {
    $_.SessionState -like $stateLookup.$SessionState
    -and ( $_.IdleTime / 60000 ) -ge $MinimumIdleMins
    -and $_.UserName -ne "$env:USERNAME"
}           
```

Comment (adamrushuk) [reply ManageRDSessions/ManageRDSessions.psm1]: I think you can remove the backticks too

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 5]
```diff
@@ -90,43 +89,60 @@ function Get-sbRDSession {
...
-        IdleTime         = ($session.IdleTime / 60000 -as [int])
-    }
-}
-} #process
+    } #process
 
-End {
-    #Intentionally empty
-}
+    End {
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: remove the End block if unused

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 6]
```diff
@@ -90,43 +89,60 @@ function Get-sbRDSession {
...
+        #Intentionally empty
+    }
 } #function
 
+# Adding a Script Method to Get-sbRDSession to allow sending messages to the session(s) found.
+$myType = "Custom.SB.RDSession"
+Update-TypeData -TypeName $myType -MemberType ScriptMethod -MemberName SendMessage -Value {
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1:123]: Mind blown!

File: tests/Get-sbRDSession.Tests.ps1 [hunk 7]
```diff
@@ -27,5 +27,17 @@ InModuleScope $ModuleName {
         It "only accepts active, idle, disconnected, connected or any" {
             { Get-sbRDSession -SessionState Dancing } | Should Throw
```
Comment (adamrushuk) [inline tests/Get-sbRDSession.Tests.ps1:28]: You can validate the scenarios that should pass too, like:
```
{ Get-sbRDSession -SessionState Active} | Should Not Throw
```
I've not tested the above, but you get the gist.

**Review by adamrushuk — CHANGES_REQUESTED** [review]
Amazing work my friend; you've come far!

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 2 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1]: Oooooooooooooo

Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1]: $false default removed

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 4 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1]: Done - it allowed the formatter to correct the indentation as predicted.

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 3 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1]: See below backtick-related comment

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 5 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1]: Done - also with unnecessary begin blocks throughout

File: tests/Get-sbRDSession.Tests.ps1 (hunk 7 above)
Comment (steevaavoo) [reply tests/Get-sbRDSession.Tests.ps1:28]: Noted and added.

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 1 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1]: Corrected throughout

**Comment by steevaavoo** [comment]
So I've made the requested changes, and once again noticed the formatter's propensity to get "funky" with the closing }s - can you see the same in my latest commit?

**Comment by steevaavoo** [comment]
Aside from that, are we looking at a release-ready module?

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 2 above)
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: This should just be `[switch]$IncludeSelf`

**Comment by adamrushuk** [comment]
almost there

**Review by adamrushuk — CHANGES_REQUESTED** [review]
Apart from the Switch $false still showing, it's looking ready.
I mentioned dynamically dot sourcing function files - instead of one big .psm1 (as shown here: https://github.com/adamrushuk/PSvCloud/blob/master/PSvCloud/PSvCloud.psm1) - but that will mess with diffs and show as new files, so I'd leave that to a separate branch and PR in future after a restructure.

**Pull request merged**